h5grove/utils.py

```diff
@@ -138,6 +136,10 @@ def sanitize_array(array: Sequence[Number], copy: bool = True) -> np.ndarray:
     return np.array(ndarray, copy=copy, order="C", dtype=_sanitize_dtype(ndarray.dtype))
 
 
+def is_numeric_array(array: np.ndarray) -> bool:
+    return np.issubdtype(array.dtype, np.number) or np.issubdtype(array.dtype, np.bool_)
```

Comment:
There is an alternative check in 
https://github.com/silx-kit/h5grove/pull/65/commits/1aded203ff3dd4f866b49985bcada9597cf66f36#diff-b4d8de5dbee9db20aaa1ee40465d7ba388f6590feef4dfbaf922acdc0712a3f9R139-R141

For a check **before** converting to numpy array rather than checking the dtype afterwards.

The advantage of this check is that in supports list of number while the other one would return `False`.

Reply:
Yup it is fine